sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-webapp/src/main/java/com/azure/spring/sample/aad/controller/ClientController.java

```diff
@@ -23,6 +26,12 @@ public String index(
         @RegisteredOAuth2AuthorizedClient("azure") OAuth2AuthorizedClient azureClient
     ) {
         model.addAttribute("userName", authentication.getName());
+        String preferredUsername = Optional.of(authentication)
```

Comment:
Could you elaborate more on the relationship between upn and prefered_username?

Reply:
@saragluna 
You can read [this doc](https://docs.microsoft.com/en-us/azure/active-directory/develop/active-directory-optional-claims) for more information about upn and preferred_username.

![image](https://user-images.githubusercontent.com/13167207/112567557-00e26e80-8e1c-11eb-8a9b-3ea4966202e7.png)

Do you mean I should add the link of [this doc](https://docs.microsoft.com/en-us/azure/active-directory/develop/active-directory-optional-claims) into the java code?